krm-functions/lib/kptrl/kptrl.go

```diff
@@ -0,0 +1,133 @@
+/*
+ Copyright 2023 Nephio.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package kptrl
+
+import (
+	"sync"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+)
+
+type ResourceList interface {
+	// AddResult add a result with error and corresponding kubeObject by
+	// appending a result to the result slice in the resourcelist
+	AddResult(err error, obj *fn.KubeObject)
```

Comment:
Trying to understand why `AddResult` takes only an error object? If we wanted only add error result, could we not use `ErrorResult` at https://github.com/GoogleContainerTools/kpt-functions-sdk/blob/master/go/fn/result.go#L274

Reply:
ErrorResult is to provide the overall result. here we are adding individual results that are seen when running the fn.